src/python/gudhi/simplex_tree.pyx

```diff
@@ -639,3 +639,10 @@ cdef class SimplexTree:
             self.thisptr = <intptr_t>(ptr.collapse_edges(nb_iter))
             # Delete old pointer
             del ptr
+
+    def __eq__(self, other:SimplexTree):
```

Comment:
I wonder if that changes the minimum version of [pc]ython we require. (I think it is fine to require it)

Reply:
Python typing involves python >= 3.5 ([source](https://docs.python.org/3/library/typing.html)) and cython >= 0.27 ([source](https://github.com/cython/cython/pull/3799/files)).

Reply:
Those requirements seem ok to me, should we document them?